File: py/indicators/core/specification.py
"""Specification contains all info needed to create an indicator."""

from .identifier import Identifier


class Specification:
    """Contains all info needed to create an indicator."""

    __slots__ = ('identifier', 'parameters', 'outputs')

    def __init__(self, identifier: Identifier, parameters: object = None,
                 outputs: list[int] | None = None) -> None:
        self.identifier = identifier
        self.parameters = parameters
        self.outputs = outputs if outputs is not None else []

File: rs/src/indicators/core/specification.rs
use super::identifier::Identifier;

/// Contains all info needed to create an indicator.
#[derive(Debug, Clone)]
pub struct Specification {
    /// Identifies the indicator to create.
    pub identifier: Identifier,
    /// Indicator-specific parameters (serialized as JSON or similar).
    pub parameters: Option<String>,
    /// Which outputs to compute (empty = all).
    pub outputs: Vec<i32>,
}

impl Specification {
    pub fn new(identifier: Identifier) -> Self {
        Self {
            identifier,
            parameters: None,
            outputs: Vec::new(),
        }
    }
}
